articles/introduction-web-assembly-continued/index.md

```diff
@@ -0,0 +1,497 @@
+        // of our playing audio buffer
+        audioBuffer.getChannelData(0).set(amplifiedAudioSamples);
+        audioBuffer.getChannelData(1).set(amplifiedAudioSamples);
+
+        createAndStartAudioBufferSource();
+    };
+
+    const pause = () => {
+        beforePlay();
+        stopAudioBufferSource();
+    };
+
+    //...
+
+}
+```
+
+We then render the following JSX:
+
+```jsx
+return (
+    <div>
+      <h1>Watch out when using headphones!!</h1>
+      <h1>Original Sine Wave</h1>
+      <div>
+        <button
+          className="original"
+          onClick={() => {
+            playOriginal();
+          }}
+        >
+          Play
+        </button>
+      </div>
+      <hr />
+      <h1>Amplified Sine Wave</h1>
+      <div>
+        <button
+          className="amplified"
+          onClick={() => {
+            playAmplified();
+          }}
+        >
+          Play
+        </button>
+      </div>
+      <hr />
+      <h1>Pause</h1>
+      <div>
+        <button
+          className="pause"
+          onClick={() => {
+            pause();
+          }}
+        >
+          Pause
+        </button>
+      </div>
+    </div>
+)
+```
+
+Finally, If you noticed, we imported some helpers functions and a constant at the very to of the App component. We need to create those tow for our app to be complete:
+
+Create a file with this path `src/helpers/index.js` and add the following code:
+
+```javascript
+export const floatSamplesToByteSamples = (floatSamples) => {
+  const byteSamples = new Uint8Array(floatSamples.length);
+  for (let i = 0; i < floatSamples.length; i++) {
+    const diff = floatSamples[i] * 127;
+    byteSamples[i] = 127 + diff;
+  }
+  return byteSamples;
+};
+
+export const byteSamplesToFloatSamples = (byteSamples) => {
+  const floatSamples = new Float32Array(byteSamples.length);
+  for (let i = 0; i < byteSamples.length; i++) {
+    const byteSample = byteSamples[i];
+    const floatSample = (byteSample - 127) / 127;
+    floatSamples[i] = floatSample;
+  }
+  return floatSamples;
+};
+
+export const generateAudioSamples = (numberOfSamples) => {
+  const audioSamples = [];
+
+  const sampleValue = 0.3;
+  for (let i = 0; i < numberOfSamples; i++) {
+    if (i < numberOfSamples / 2) {
+      audioSamples[i] = sampleValue;
+    } else {
+      audioSamples[i] = sampleValue * -1;
+    }
+  }
+  return audioSamples;
+};
+```
+
+This one is a bit unnecessary since we only have on constant, but it shows some good practices in terms of a React Application project structure.
+
+`src/constants/index.js` :
+
+```javascript
+export const NUMBER_OF_SAMPLES = 1024;
+```
+
+This marks the end of our demo application, and if you followed all the steps above you should end up with something similar to [this](https://github.com/lucasm08/react-wasm-audio-rendering). Now, all that's left  is the start the app with `npm start`:
+
+![Web browser screenshot](browser-screenshot.png "")
```

Comment:
@lucasm08  - just wanted to check if there is an image missing from the article?

Reply:
Nope! it's my mistake. I forgot to update the markdown after editing the image file name. oops!!

Reply:
I fixed it. Everything should alright now